### ISSUE TITLE
HOCON ignores environment overrides inside of Docker Image

### ISSUE THREAD

**Issue by KaoruDev** [softwaremill/elasticmq#500]
# Problem
I'm trying to set up a docker compose file to run elasticmq in two different scenarios. TLDR: the problem is that one scenario requires the hostname in the response of a `GetQueueUrl` request to be `localhost` and another scenario requires it be the container name, `elasticmq`. I tried to set this up by setting an ENV override in the `conf` file like so:

```
node-address {
  protocol = http
  host = elasticmq
  host = ${?ELASTICMQ_NODE_ADDRESS_HOST}
  port = 9324
  context-path = ""
}
```

Alas, it appears that `ELASTICMQ_NODE_ADDRESS_HOST` is not picked up by `HOCON` at all. Yet when I override the `entrypoint` to `printenv` I can see it:

```
elasticmq    | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
elasticmq    | HOSTNAME=elasticmq
elasticmq    | ELASTICMQ_NODE_ADDRESS_HOST=foobar
elasticmq    | HOME=/home/demiourgos728
elasticmq exited with code 0
```

# Steps to reproduce
Setup conf file:

```
include classpath("application.conf")

default-queue-config {
  defaultVisibilityTimeout = 10 seconds
  delay = 5 seconds
  receiveMessageWait = 0 seconds
  fifo = false
  contentBasedDeduplication = false
}

node-address {
  protocol = http
  host = localhost
  host = ${?ELASTICMQ_NODE_ADDRESS_HOST}
  port = 9324
  context-path = ""
}

queues {
  sum-queue: ${default-queue-config}
}

aws {
  region = us-west-2
}
```

Docker compose file:

```yaml
---
version: '3.2'
services:
  elasticmq:
    container_name: elasticmq
    hostname: elasticmq
    image: softwaremill/elasticmq-native:latest
    volumes:
      - ./configs/elasticmq.conf:/opt/elasticmq.conf # path to file above
    ports:
      - "9324:9324"
    expose:
      - "9324"
    environment:
      - ELASTICMQ_NODE_ADDRESS_HOST=foobar
...
```

Run: `curl "localhost:9324?Action=GetQueueUrl&QueueName=sum-queue"`

Expect:

```xml
<GetQueueUrlResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/">
  <GetQueueUrlResult>
    <QueueUrl>http://foobar:9324/000000000000/sum-queue</QueueUrl>
  </GetQueueUrlResult>
  <ResponseMetadata>
    <RequestId>00000000-0000-0000-0000-000000000000</RequestId>
  </ResponseMetadata>
</GetQueueUrlResponse>
```

Get:

```xml
<GetQueueUrlResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/">
  <GetQueueUrlResult>
    <QueueUrl>http://localhost:9324/000000000000/sum-queue</QueueUrl>
  </GetQueueUrlResult>
  <ResponseMetadata>
    <RequestId>00000000-0000-0000-0000-000000000000</RequestId>
  </ResponseMetadata>
</GetQueueUrlResponse>
```

**Comment by KaoruDev** [comment]
I was able to get this working as expected by creating my own docker image by following instructions in the `README.md`. Still I think this should be looked at, or at the very least explicitly called out in the `README.md`

**Comment by pierscin** [comment]
I've tried `run` with the environment variable, `env-file` through command line and file and also changing your `compose` - none of that fixed the issue, but as you've said - variables are present in the container. The only thing I've noticed is that this issue is only present when using native image - when I've changed it to full JVM it worked. Also tests were fine.

After some debugging I've just invalidated cache of `ConfigLoader`, because variables were there and everything we use is up-to-date. Now it works perfectly. 🪄 

Thanks for letting us know 👍
